program-runtime/src/loaded_programs.rs

```diff
@@ -772,31 +780,52 @@ impl LoadedPrograms {
                         }
                     }
                 }
-                missing.push((key, count));
+                missing.push((key, count, false));
                 None
             })
             .collect::<HashMap<Pubkey, Arc<LoadedProgram>>>();
-
         self.stats
             .misses
             .fetch_add(missing.len() as u64, Ordering::Relaxed);
         self.stats
             .hits
             .fetch_add(found.len() as u64, Ordering::Relaxed);
-        ExtractedPrograms {
-            loaded: LoadedProgramsForTxBatch {
-                entries: found,
-                slot: working_slot.current_slot(),
-                environments: environments.clone(),
-            },
-            missing,
-            unloaded,
+        for (key, count, reload) in missing.into_iter() {
+            let entry = self.assign_program(
+                key,
+                Arc::new(LoadedProgram::new_loading(loaded.clone(), count)),
+            );
+            self.cooperative_loading_queue
+                .push_back((key, entry, reload));
         }
+        loaded.lock().unwrap().entries = found;
+        loaded
+    }
+
+    /// In cooperative loading a TX batch calls this to load the next task
+    pub fn get_cooperative_loading_task(&mut self) -> Option<(Pubkey, Arc<LoadedProgram>, bool)> {
+        self.cooperative_loading_queue.pop_front()
+    }
+
+    /// Upon finishing a task in cooperative loading a TX batch calls this to store the result
+    pub fn submit_cooperative_loading_task(
```

Comment:
I think this should be called cooperative_loading_task_complete. Submit makes it
sound like it's _adding_ more work